workspaces/arborist/lib/yarn-lock.js

```diff
@@ -33,10 +32,45 @@ const consistentResolve = require('./consistent-resolve.js')
 const { dirname } = require('path')
 const { breadth } = require('treeverse')
 
+// Sort Yarn entries respecting the yarn.lock sort order
+const yarnEntryPriorities = {
+  name: 1,
+  version: 2,
+  uid: 3,
+  resolved: 4,
+  integrity: 5,
+  registry: 6,
+  dependencies: 7,
+}
+
+const priorityThenLocaleCompare = (a, b) => {
+  if (yarnEntryPriorities[a] || yarnEntryPriorities[b]) {
+    return (yarnEntryPriorities[a] || 100) > (yarnEntryPriorities[b] || 100) ? 1 : -1
+  }
+
+  return localeCompare(a, b)
+}
+
+const shouldQuoteString = str => {
+  return str.indexOf('true') === 0 ||
+      str.indexOf('false') === 0 ||
+      /[:\s\n\\",[\]]/g.test(str) ||
+      /^[0-9]/g.test(str) ||
+      !/^[a-zA-Z]/g.test(str)
```

Comment:
If it starts with 0-9 then it doesn't start w/ a-zA-Z so the 0-9 check isn't needed.